### DISCUSSION TITLE
FlutterError (RenderCustomMultiChildLayoutBox object was given an infinite size during layout.

### DISCUSSION THREAD

**Discussion by badriyahaddad** [community/community#145264] [Programming Help]
### Select Topic Area

Question

### Body

so I have this code that causes this exception :
the code:
import 'package:flutter/material.dart';
import 'package:petme/presentation/widgets/clickup_widget/filter_buttons.dart';

class DiscoveryScreen extends StatelessWidget {
  const DiscoveryScreen({super.key});

  @override
  Widget build(BuildContext context) {
    Size size = MediaQuery.of(context).size;

    return SingleChildScrollView(
      child: Scaffold(
        body: Column(children: [
          Expanded(
              child: SizedBox(
                  height: size.height / 12, child: const FilterButtons())),
        ]),
      ),
    );
  }
}
the FilterButtons widget:
import 'package:flutter/material.dart';
import 'package:petme/core/theme/colors.dart';
import 'package:petme/data/fake_data.dart';
import 'package:petme/presentation/widgets/clickup_widget/main_button.dart';

class FilterButtons extends StatefulWidget {
  const FilterButtons({super.key});

  @override
  // ignore: library_private_types_in_public_api
  _FilterButtonsState createState() => _FilterButtonsState();
}

class _FilterButtonsState extends State<FilterButtons> {
  int selectedIndex = -1;

  bool isMainButtonDisabled = false;

  @override
  Widget build(BuildContext context) {
    Size size = MediaQuery.of(context).size;
    return Scaffold(
      body: SizedBox(
        height: size.height / 23,
        width: size.width,
        child: ListView.builder(
          shrinkWrap: true,
          scrollDirection: Axis.horizontal,
          itemCount: filterDiscTxt.length,
          itemBuilder: (context, index) {
            return Padding(
              padding: const EdgeInsets.symmetric(horizontal: 8.0),
              child: MainButton(
                isMain: selectedIndex == index,
                isIcon: false,
                isdisabled: false,
                isWide: false,
                clr: AppColors.secondaryColor,
                txt: filterDiscTxt[index]['txt'],
                txtClr: Colors.white,
                action: () {
                  setState(() {
                    selectedIndex = index;
                    isMainButtonDisabled = true;
                  });
                },
                isHightNeeded: false,
                isWidthNeeded: true,
                height: 0,
                width: size.width / 5,
                isTextStyleNeeded: false,
                txtStyle: null,
                icon: null,
              ),
            );
          },
        ),
      ),
    );
  }
}
the exception:
Exception has occurred.
FlutterError (RenderCustomMultiChildLayoutBox object was given an infinite size during layout.
This probably means that it is a render object that tries to be as big as possible, but it was put inside another render object that allows its children to pick their own size.
The nearest ancestor providing an unbounded height constraint is: _RenderSingleChildViewport#10e87 relayoutBoundary=up9 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
  needs compositing
  creator: _SingleChildViewport ← IgnorePointer-[GlobalKey#7d407] ← Semantics ← Listener ← _GestureSemantics ← RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#39632] ← Listener ← _ScrollableScope ← _ScrollSemantics-[GlobalKey#76cf7] ← NotificationListener<ScrollMetricsNotification> ← Transform ← ClipRect ← ⋯
  parentData: <none> (can use size)
  constraints: BoxConstraints(0.0<=w<=411.4, 0.0<=h<=661.4)
  size: MISSING
  offset: Offset(0.0, -0.0)
The constraints that applied to the RenderCustomMultiChildLayoutBox were:
  BoxConstraints(0.0<=w<=411.4, 0.0<=h<=Infinity)
The exact size it was given was:
  Size(411.4, Infinity)
See https://flutter.dev/to/unbounded-constraints for more information.)

**Comment by stevepopovich** [comment]
Hi @badriyahaddad seems like you are having issues building a Flutter app. This forum is for feedback about GitHub. I would try posting something like this to StackOverflow, thanks.

**Reply by queenofcorgis** [reply]
We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
